pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java

```diff
@@ -400,6 +400,11 @@ public boolean registerNamespace(String namespace, boolean ensureOwned) throws P
     private void searchForCandidateBroker(NamespaceBundle bundle,
                                           CompletableFuture<Optional<LookupResult>> lookupFuture,
                                           LookupOptions options) {
+        if( null == pulsar.getLeaderElectionService() || ! pulsar.getLeaderElectionService().isElected()) {
+            LOG.warn("The leader election has not yet been completed! NamespaceBundle[{}]", bundle);
+            lookupFuture.completeExceptionally(new IllegalStateException("The leader election has not yet been completed!"));
```

Comment:
Thanks for the great contribution, I think it's better to return a retryable exception to the client? So that the client can reconnect later. Does this make sense?

Reply:
As discussed with @4onni, this will resultin a client lookup timeout, and the client will relookup later, so it's not a problem here.